### PR TITLE
Add endpoints for creating and sending voucher payments

### PR DESCRIPTION
**Pull request by akheron**
#### Summary

While at it, also make invoice number series configurable (Oulu needs this).

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/payments/PaymentController.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    data class SendPaymentsRequest(
+        val paymentDate: LocalDate,
+        val dueDate: LocalDate,
+        val paymentIds: List<PaymentId>,
+    )
+
+    @PostMapping("/send")
+    fun sendInvoices(
```
Comment (jaahon) [inline service/src/main/kotlin/fi/espoo/evaka/payments/PaymentController.kt:37]: Should this be `sendPayments`? 😄

File: service/src/main/kotlin/fi/espoo/evaka/payments/PaymentService.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    ) {
+        val seriesStart = featureConfig.paymentNumberSeriesStart ?: throw Error("paymentNumberSeriesStart not configured")
+        val payments = tx.readPaymentsByIds(paymentIds)
+
+        val notDrafts = payments.filterNot { it.status == PaymentStatus.DRAFT }
+        if (notDrafts.isNotEmpty()) {
+            throw BadRequest("Some payments are not drafts")
+        }
+
+        val units = tx.getPaymentUnitsByIds(payments.map { it.unit.id }).associateBy { it.id }
```
Comment (jaahon) [inline service/src/main/kotlin/fi/espoo/evaka/payments/PaymentService.kt:37]: This is not a big deal, but why not query the necessary unit data in the same query with the other payment data?

**Review by jaahon — COMMENTED** [review]
I don't know if you thought of this, but is there any sense in storing the drafts? Couldn't they just be generated dynamically for front end? You wouldn't need the complicated check constraint then either.

File: service/src/main/resources/db/migration/V243__payment.sql [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    id uuid PRIMARY KEY NOT NULL DEFAULT ext.uuid_generate_v1mc(),
+    created timestamp with time zone NOT NULL DEFAULT now(),
+    updated timestamp with time zone NOT NULL DEFAULT now(),
+    unit_id uuid NOT NULL CONSTRAINT fk$unit REFERENCES daycare (id),
+    unit_name text NOT NULL,
+    unit_business_id text,
+    unit_iban text,
+    unit_provider_id text,
+    period daterange NOT NULL,
+    number bigint,
```
Comment (Gekkio) [inline service/src/main/resources/db/migration/V243__payment.sql:13]: Unique constraint?

File: service/src/main/resources/db/migration/V243__payment.sql [hunk 4]
```diff
@@ -0,0 +1,35 @@
+CREATE TYPE payment_status AS ENUM ('DRAFT', 'SENT');
+
+CREATE TABLE payment (
+    id uuid PRIMARY KEY NOT NULL DEFAULT ext.uuid_generate_v1mc(),
+    created timestamp with time zone NOT NULL DEFAULT now(),
+    updated timestamp with time zone NOT NULL DEFAULT now(),
+    unit_id uuid NOT NULL CONSTRAINT fk$unit REFERENCES daycare (id),
```
Comment (Gekkio) [inline service/src/main/resources/db/migration/V243__payment.sql:7]: Index?

File: service/src/main/resources/db/migration/V243__payment.sql [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    unit_iban text,
+    unit_provider_id text,
+    period daterange NOT NULL,
+    number bigint,
+    amount int NOT NULL,
+    status payment_status NOT NULL,
+    payment_date date,
+    due_date date,
+    sent_at timestamp with time zone,
+    sent_by uuid CONSTRAINT fk$sent_by REFERENCES evaka_user (id),
```
Comment (Gekkio) [inline service/src/main/resources/db/migration/V243__payment.sql:19]: Index?

**Comment by akheron** [comment]
Superseded by https://github.com/espoon-voltti/evaka/pull/2582

File: service/src/main/resources/db/migration/V243__payment.sql (hunk 3 above)
Comment (akheron) [reply service/src/main/resources/db/migration/V243__payment.sql:13]: Added in #2582

File: service/src/main/resources/db/migration/V243__payment.sql (hunk 4 above)
Comment (akheron) [reply service/src/main/resources/db/migration/V243__payment.sql:7]: Added in #2582. I didn't realize that this syntax doesn't add an index.

File: service/src/main/resources/db/migration/V243__payment.sql (hunk 5 above)
Comment (akheron) [reply service/src/main/resources/db/migration/V243__payment.sql:19]: Ditto

File: service/src/main/kotlin/fi/espoo/evaka/payments/PaymentService.kt (hunk 2 above)
Comment (akheron) [reply service/src/main/kotlin/fi/espoo/evaka/payments/PaymentService.kt:37]: Good idea, fixed in #2582

**Pull request closed without merging**